### ISSUE TITLE
415607E1 - Call of Duty 3

### ISSUE THREAD

**Issue by 1HackJack1** [xenia-canary/game-compatibility#27]
[Marketplace](https://marketplace.xbox.com/en-US/Product/Call-of-Duty-3/66acd000-77fe-1000-9115-d802415607e1)

Tested on https://github.com/xenia-canary/xenia-canary/commit/293e9582ec35d30ef17a25f08858a82eed945f69

# Issues:
Menu, intro etc. - great works. The game has problems with rendering textures(Corrupt Drawing)(Drawing ground, trees, etc.), namely "horizontal lines." The sound works well. The game itself can "freeze" anywhere in the 1-st mission(Help only restart emulator), sometimes happens error "Graphic device lost". 1-st mission playing until we are getting up at truck, then emulator "freezes" (Other missions works, but after few second emulator "freezes").

# Log:
[xenia.zip](https://github.com/xenia-canary/game-compatibility/files/4198176/xenia.zip)

# Screenshot(s):
![XeniaCanary_Call of Duty 3_13_02_2020](https://user-images.githubusercontent.com/60978912/74434127-6c60bb00-4e6a-11ea-85ba-6b269c52fddc.png)

![XeniaCanary_Call of Duty 3_13_02_2020_2](https://user-images.githubusercontent.com/60978912/74434141-71be0580-4e6a-11ea-88c7-c054eb6bcc14.png)

![XeniaCanary_Call of Duty 3_13_02_2020_3](https://user-images.githubusercontent.com/60978912/74434152-75518c80-4e6a-11ea-9111-fab31a83a29b.png)

![XeniaCanary_Call of Duty 3_13_02_2020_4](https://user-images.githubusercontent.com/60978912/74434157-784c7d00-4e6a-11ea-89ce-e13009725223.png)

![XeniaCanary_Call of Duty 3_13_02_2020_5](https://user-images.githubusercontent.com/60978912/74436911-ec3d5400-4e6f-11ea-8189-08eb9226cc03.png)

# Labels:
state-gameplay, gpu-corrupt-drawing, gpu-missing-drawing, tech-middleware-demonware

**Comment by marinesciencedude** [comment]
Tested on https://github.com/xenia-canary/xenia-canary/commit/97fdf9c6dd9a6bad00182714b9205ef32e40d4f4/

### Issues

As reported by [FragsAreUs at xenia master game comaptibility](https://github.com/xenia-project/game-compatibility/issues/1320#issuecomment-1248637023), multiplayer maps now load, and light sources such as those on Fuel Plant are rendered through map geometry. Could not confirm crashing after 5-10 minutes.
Main menu music cuts itself short and loops back to the start

### Log
Singleplayer Main Menu: [xenia.zip](https://github.com/xenia-canary/game-compatibility/files/10222179/xenia.zip)
Multiplayer Main Menu: [xenia.zip](https://github.com/xenia-canary/game-compatibility/files/10222180/xenia.zip)

### Screenshots
![FuelPlant](https://user-images.githubusercontent.com/14031281/207445924-83f7320e-81d1-4fa0-b0a6-5db22ed52ec3.png)
![FuelPlantLights](https://user-images.githubusercontent.com/14031281/207445930-3fcde9da-8f52-4014-8a91-72ba98c1f571.png)

### Videos
https://user-images.githubusercontent.com/14031281/207445870-978b2dac-0e66-4161-a1a3-3f41b4247fac.mp4

https://user-images.githubusercontent.com/14031281/207445664-79738099-8a60-436f-a3f0-3692affb5943.mp4

Extended black video at the start while audio is playing is a lapse on the recording software's part

**Comment by VladWinner** [comment]
Patches: https://github.com/xenia-canary/game-patches/pull/329

**Comment by kimboslice99** [comment]
Can confirm the patch works as intended, so this can probably be closed

**Comment by VladWinner** [comment]
I disagree. These patches are dirty hacks that remove grass and change DOF radius, rather than fixing it.

**Comment by kimboslice99** [comment]
Ah, fair point. I figured your intent with posting the patch PR was that this was considered the fix.

Thanks for clearing that up for me

**Comment by heroKAYAR** [comment]
![Image](https://github.com/user-attachments/assets/8f5b82bf-e567-4bf8-92f2-d85d14155309)
![Image](https://github.com/user-attachments/assets/5eea369e-fe08-470a-8650-ddbc82593736)
![Image](https://github.com/user-attachments/assets/d27ed9a3-50b1-4c0d-b18c-33f8d812d205)
![Image](https://github.com/user-attachments/assets/21534f36-7b7a-4d12-9229-461eb4c0dd5d)

I have a problem I’m getting blinking/flickering.
I tried all patches and right settings ,  (installed COD 3 TU3 too)
My draw setting is 2 — maybe that’s the problem?

**Comment by A1eNaz** [comment]
> ![Image](https://github.com/user-attachments/assets/8f5b82bf-e567-4bf8-92f2-d85d14155309) ![Image](https://github.com/user-attachments/assets/5eea369e-fe08-470a-8650-ddbc82593736) ![Image](https://github.com/user-attachments/assets/d27ed9a3-50b1-4c0d-b18c-33f8d812d205) ![Image](https://github.com/user-attachments/assets/21534f36-7b7a-4d12-9229-461eb4c0dd5d)
> 
> I have a problem I’m getting blinking/flickering. I tried all patches and right settings , (installed COD 3 TU3 too) My draw setting is 2 — maybe that’s the problem?

don't upscale
i.e. set `draw_resolution_scale to 1`

**Comment by KenJinWong** [comment]
![Image](https://github.com/user-attachments/assets/5719e53c-ec32-40e7-a01c-504624d7989d)
the tank just got stuck from the door

**Comment by KenJinWong** [comment]
![Image](https://github.com/user-attachments/assets/804c38cf-8816-4acf-821f-969992f6452e)

![Image](https://github.com/user-attachments/assets/ff5ca242-f2e6-4e9a-85f3-f9d71f270d8d)
How do i fix that bug?
The tank have stuck the door 
I always load the checkpoint and its still not working

**Comment by nfxw** [comment]
Move to playable.

![Image](https://github.com/user-attachments/assets/bb66a5f9-e7b4-4f93-b5eb-a0fb4cf4042a)
![Image](https://github.com/user-attachments/assets/2d1fc185-0b0b-4c68-9f29-ba23d5a7a57e)

**Comment by TGP482** [comment]
> don't upscale i.e. set `draw_resolution_scale to 1`

The resolution scale issues in this game and others with the screen flickering have been fixed recently on Xenia Edge. I played through the whole game using 4x resolution scale with the FPS hovering around 140.

**Comment by Clippy95** [comment]
> > don't upscale i.e. set `draw_resolution_scale to 1`
> 
> The resolution scale issues in this game and others with the screen flickering have been fixed recently on Xenia Edge. I played through the whole game using 4x resolution scale with the FPS hovering around 140.

The upscaling fixes were merged into Canary and I believe SP upscales fine now in canary, in MP it always worked though

**Comment by BorealisEcho** [comment]
As of fbd620c_canary_experimental (2026-05-30) release, this game benefits from ZPD occlusion queries.

Set `occlusion_query = "fast"` or `occlusion_query = "strict"` to fix unoccluded lens flares.

~~Game still needs the other patches mentioned above.~~ Grass issue is fixed

**Comment by goldislead** [comment]
The grass rendering bug is tied to VS `4B4E6FDA02FB39C3`. The shader gives some vertices in each grass strip a position `w` of 0 while others, including the roots, get 1. Forcing that value to 1 makes the grass more coherent, though not exactly console accurate.

Also, it's not using ZPD at all for corona occlusion, using a depth resolve instead. `readback_resolve = "fast"` or `readback_resolve = "full"` _mostly_ fixes corona occlusion, but again, it's not console accurate.

Grass draw distance seems to be handled via VIZ conditional rendering.

**Comment by michaeloliverx** [comment]
Found the root cause of the grass rendering bug https://github.com/xenia-canary/xenia-canary/pull/1149
